File: backend/ml-service/app/schemas/prediction_request.py
from pydantic import BaseModel, Field, ConfigDict
from typing import Optional

class PredictionRequest(BaseModel):
    # Enable alias support for your hyphenated column names
    model_config = ConfigDict(populate_by_name=True)

    # Required fields (the '...' remains implicitly required for these)
    RevolvingUtilizationOfUnsecuredLines: float
    age: int
    NumberOfTime30_59DaysPastDueNotWorse: int = Field(alias="NumberOfTime30-59DaysPastDueNotWorse")
    NumberOfTime60_89DaysPastDueNotWorse: int = Field(alias="NumberOfTime60-89DaysPastDueNotWorse")
    DebtRatio: float
    NumberOfOpenCreditLinesAndLoans: int
    NumberOfTimes90DaysLate: int
    NumberRealEstateLoansOrLines: int
    
    # Optional fields (Now flexible!)
    MonthlyIncome: Optional[float] = None
    NumberOfDependents: Optional[float] = None

File: backend/core-service/src/main/java/com/justifai/coreservice/dto/PredictionRequest.java
package com.justifai.coreservice.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

public record PredictionRequest(
                @JsonProperty("RevolvingUtilizationOfUnsecuredLines") Double RevolvingUtilizationOfUnsecuredLines,
                @JsonProperty("age") Integer age,
                @JsonProperty("NumberOfTime30-59DaysPastDueNotWorse") Integer NumberOfTime30_59DaysPastDueNotWorse,
                @JsonProperty("NumberOfTime60-89DaysPastDueNotWorse") Integer NumberOfTime60_89DaysPastDueNotWorse,
                @JsonProperty("DebtRatio") Double DebtRatio,
                @JsonProperty("NumberOfOpenCreditLinesAndLoans") Integer NumberOfOpenCreditLinesAndLoans,
                @JsonProperty("NumberOfTimes90DaysLate") Integer NumberOfTimes90DaysLate,
                @JsonProperty("NumberRealEstateLoansOrLines") Integer NumberRealEstateLoansOrLines,
                @JsonProperty("MonthlyIncome") Double MonthlyIncome,
                @JsonProperty("NumberOfDependents") Double NumberOfDependents) {
}
